File: c/xor.c
#include <stdio.h>

void swap(char *a, char *b) {
  *a ^= *b;
  *b ^= *a;
  *a ^= *b;
}

int main(int argc, char* argv[]) {
  char a = 7;
  char b = 3;
  printf("A=%d and B=%d\n", a, b);
  swap(&a,&b);
  printf("A=%d and B=%d\n", a, b);
}

File: python/xor.py
def swap(A,B):
  A = A^B
  B = A^B
  A = A^B
  return A,B

A = 7
B = 4
print("A=%d B=%d" % (A,B))
A,B = swap(A,B)
print("A=%d B=%d" % (A,B))

